source/particle/property/integrated_strain.cc

```diff
@@ -43,7 +43,7 @@ namespace aspect
                                                       const std::vector<Tensor<1,dim>> &gradients,
                                                       typename ParticleHandler<dim>::particle_iterator &particle) const
       {
-        auto &data = particle->get_properties();
+        const auto data = particle->get_properties();
```

Comment:
`get_properties()` used to return a `const ArrayView<...>` object. It isn't a reference, so capturing it with `auto &` was always a poor design choice. It worked because the compiler deduced `auto = const ArrayView`, and binding a temporary to a const reference is allowed. But https://github.com/dealii/dealii/pull/15671 changed the return type to drop the `const`, which means that the compiler deduces `auto = ArrayView`, and now we have a non-const reference for which the returned temporary is no longer able to bind -- and the compiler errors out.